master/internal/user/postgres_users.go

```diff
@@ -46,74 +45,52 @@ func StartUserSession(ctx context.Context, user *model.User) (string, error) {
-	_, err := tx.NewDelete().Model(&model.AgentUserGroup{}).Where("user_id = ?", userID).Exec(ctx)
+	_, err := tx.NewDelete().
+		Model(&model.AgentUserGroup{}).
+		Where("user_id = ?", userID).Exec(ctx)
 	if err != nil {
-		return errors.WithStack(err)
+		return err
 	}
 	return nil
 }
 
-// DeleteUserSessionByToken deletes user session if found
+// DeleteSessionByToken deletes user session if found
 // (externally managed sessions are not stored in the DB and will not be found).
-func DeleteUserSessionByToken(ctx context.Context, token string) error {
+func DeleteSessionByToken(ctx context.Context, token string) error {
 	v2 := paseto.NewV2()
 	var session model.UserSession
 	// verification will fail when using external token (Jwt instead of Paseto)
 	if err := v2.Verify(token, db.GetTokenKeys().PublicKey, &session, nil); err != nil {
 		return nil
 	}
-	return DeleteUserSessionByID(ctx, session.ID)
+	return DeleteSessionByID(ctx, session.ID)
 }
 
-// DeleteUserSessionByID deletes the user session with the given ID.
-func DeleteUserSessionByID(ctx context.Context, sessionID model.SessionID) error {
-	_, err := db.Bun().NewDelete().Model(&model.UserSession{}).Where("id = ?", sessionID).Exec(ctx)
+// DeleteSessionByID deletes the user session with the given ID.
+func DeleteSessionByID(ctx context.Context, sessionID model.SessionID) error {
+	_, err := db.Bun().NewDelete().
+		Model(&model.UserSession{}).
+		Where("id = ?", sessionID).
+		Exec(ctx)
 	return err
 }
 
+// addAgentUserGroup UPSERT's the existence of an agent user group.
 func addAgentUserGroup(
 	ctx context.Context,
+	idb bun.IDB,
 	userID model.UserID,
 	ug *model.AgentUserGroup,
 ) error {
-	tx, err := db.Bun().BeginTx(ctx, nil)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		txErr := tx.Rollback()
-		if txErr != nil && txErr != sql.ErrTxDone {
-			log.WithError(txErr).Error("error rolling back transaction in addUserExec")
-		}
-	}()
-
 	next := *ug
 	next.UserID = userID
-	if _, err = tx.NewInsert().Model(next).Exec(ctx); err != nil {
-		return errors.WithStack(err)
-	}
-
-	if err = tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing changes in addAgentUserGroup")
-	}
-
-	return nil
+	_, err := idb.NewInsert().Model(next).Exec(ctx)
+	return err
 }
 
-// addUserExec execs an INSERT to create a new user.
-func addUserExec(ctx context.Context, user *model.User) (model.UserID, error) {
-	tx, err := db.Bun().BeginTx(ctx, nil)
-	if err != nil {
-		return 0, err
-	}
-	defer func() {
-		txErr := tx.Rollback()
-		if txErr != nil && txErr != sql.ErrTxDone {
-			log.WithError(txErr).Error("error rolling back transaction in addUserExec")
-		}
-	}()
-
-	_, err = tx.NewInsert().Model(user).ExcludeColumn("id").Returning("*").Exec(ctx)
+// addExec execs an UPSERT the existence of a new user.
+func addExec(ctx context.Context, idb bun.IDB, user *model.User) (model.UserID, error) {
+	_, err := idb.NewInsert().Model(user).ExcludeColumn("id").Returning("*").Exec(ctx)
```

Comment:
i think we dont need the `ExcludeColumn("id")` if we mark the user id as autoincrement,pk but i could be wrong, not super familiar with bun.